File: app/src/main/java/nl/jpelgrm/movienotifier/data/DaoUsers.java
package nl.jpelgrm.movienotifier.data;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;
import nl.jpelgrm.movienotifier.models.User;

@Dao
public interface DaoUsers {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void add(User user);

    @Query("SELECT * FROM Users")
    LiveData<List<User>> getUsers();

    @Query("SELECT * FROM Users")
    List<User> getUsersSynchronous();

    @Query("SELECT * FROM Users WHERE ID = :id")
    LiveData<User> getUserById(String id);

    @Update
    void update(User user);

    @Delete
    void delete(User user);
}

File: app/src/main/java/nl/jpelgrm/movienotifier/data/DaoUsers.kt
package nl.jpelgrm.movienotifier.data

import androidx.lifecycle.LiveData
import androidx.room.*
import nl.jpelgrm.movienotifier.models.User

@Dao
interface DaoUsers {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun add(user: User)

    @get:Query("SELECT * FROM Users")
    val users: LiveData<List<User>>

    @get:Query("SELECT * FROM Users")
    val usersSynchronous: List<User>

    @Query("SELECT * FROM Users WHERE ID = :id")
    fun getUserById(id: String): LiveData<User>

    @Query("SELECT * FROM Users WHERE ID = :id")
    fun getUserByIdSynchronous(id: String): User

    @Update
    fun update(user: User)

    @Delete
    fun delete(user: User)
}